packages/plugin-core/src/test/suite-integ/NoteLookupCommand.test.ts

```diff
@@ -749,6 +752,433 @@ suite("NoteLookupCommand", function () {
       });
     });
 
+    describeMultiWS(
+      "WHEN schema template references to a template note that lies in a different vault",
+      {
+        ctx,
+        preSetupHook: async ({ wsRoot, vaults }) => {
+          await ENGINE_HOOKS_MULTI.setupBasicMulti({ wsRoot, vaults });
+          // Template is in vault2
+          await NoteTestUtilsV4.createNote({
+            wsRoot,
+            body: "food ch2 template",
+            fname: "template.ch2",
+            vault: vaults[1],
+          });
+        },
+        postSetupHook: async ({ wsRoot, vaults }) => {
+          // Schema is in vault1
+          const vault = vaults[0];
+          await NoteTestUtilsV4.createSchema({
+            fname: "food",
+            wsRoot,
+            vault,
+            modifier: (schema) => {
+              const schemas = [
+                SchemaUtils.createFromSchemaOpts({
+                  id: "food",
+                  parent: "root",
+                  fname: "food",
+                  children: ["ch2"],
+                  vault,
+                }),
+                SchemaUtils.createFromSchemaRaw({
+                  id: "ch2",
+                  template: {
+                    id: "template.ch2",
+                    type: "note",
+                  },
+                  namespace: true,
+                  vault,
+                }),
+              ];
+              schemas.map((s) => {
+                schema.schemas[s.id] = s;
+              });
+              return schema;
+            },
+          });
+        },
+      },
+      () => {
+        test("THEN template body gets applied to new note", async () => {
+          const cmd = new NoteLookupCommand();
+          await cmd.run({
+            initialValue: "food.ch2",
+            noConfirm: true,
+          });
+          const { engine, vaults } = ExtensionProvider.getDWorkspace();
+
+          const newNote = NoteUtils.getNoteByFnameFromEngine({
+            fname: "food.ch2",
+            engine,
+            vault: vaults[0],
+          });
+          expect(_.trim(newNote?.body)).toEqual("food ch2 template");
+        });
+      }
+    );
+
+    describeMultiWS(
+      "WHEN schema template references to a template note that lies in a different vault using xvault notation",
+      {
+        ctx,
+        preSetupHook: async ({ wsRoot, vaults }) => {
+          await ENGINE_HOOKS_MULTI.setupBasicMulti({ wsRoot, vaults });
+          // Template is in vault2
+          await NoteTestUtilsV4.createNote({
```

Comment:
Can we add another note with the same fname and a different body to a different vault, so we can test that it's actually finding the right template note?

Reply:
This should be covered in the subsequent test case "WHEN schema template references to a template note that lies in multiple vaults". In this case, user is prompted to pick the right vault

Reply:
That's a different test. I mean if the template note is in the form of `dendron://vault1/note` and there are multiple notes with the same name in different vaults, then the code should automatically pick the one in `vault1`.

This test is already testing that, it's just that there are no other notes with the same name so the test would pass even if the template was set to `note' without the vault specified.

Reply:
Ah I see what you mean. Yea that makes sense, added note